saleor/account/throttling.py

```diff
@@ -0,0 +1,144 @@
+import logging
+from datetime import timedelta
+from math import ceil
+from typing import Optional
+
+from django.core.cache import cache
+from django.core.exceptions import ValidationError
+from django.utils import timezone
+
+from ..core.utils import get_client_ip
+from . import models
+from .error_codes import AccountErrorCode
+from .utils import retrieve_user_by_email
+
+MIN_DELAY = 1
+MAX_DELAY = 3600
+ATTEMPT_COUNTER_EXPIRE_TIME = 7200
+
+logger = logging.getLogger(__name__)
+
+
+def authenticate_with_throttling(request, email, password) -> Optional[models.User]:
+    ip = get_client_ip(request)
+    if not ip:
+        logger.warning("Unknown request's IP address.")
+        raise ValidationError(
+            {
+                "email": ValidationError(
+                    "Can't indentify requester IP address.",
+                    code=AccountErrorCode.UNKNOWN_IP_ADDRESS.value,
+                )
+            }
+        )
+
+    ip_key = get_cache_key_failed_ip(ip)
+    block_key = get_cache_key_blocked_ip(ip)
+
+    # block the IP address before the next attempt to prevent concurrent requests
+    if add_block(block_key, MIN_DELAY) is False:
+        next_attempt_time = cache.get(block_key) or timezone.now() + timedelta(
+            seconds=MIN_DELAY
+        )
+        raise ValidationError(
+            {
+                "email": ValidationError(
+                    f"Logging has been suspended till {next_attempt_time}.",
```

Comment:
Same question as above. Is the error related to the multiple logins for the same email or related to blocked IP? If related to IP I would not "connect" it with `email` field. 
---
Additionally, would be good to provide in the message more info why we blocked the login process. Like to many requests from the same IP, or too many request for the same email etc

Reply:
> Two concurrent login attempts
> Login blocked due to too many failed attempts

These two means in general too many attempts right? It can be too many attempts to log in for single user, or for single IP. So maybe adding too many attempts in general would be enough?
So something like:
`Too many attempts to log-in. Logging has been suspended till {next_attempt_time}`
🤔